src/Compilers/CSharp/Test/Symbol/Symbols/LocalFunctionTests.cs

```diff
@@ -189,5 +189,116 @@ public void M()
+
+                public class C
+                {
+                }
+
+                public static class Extensions
+                {
+                    public static string MyExtension(this C c)
+                        => string.Empty;
+                }
+                """).VerifyDiagnostics(
+                    // (7,7): error CS8421: A static local function cannot contain a reference to 'c'.
+                    //     x(c);
+                    Diagnostic(ErrorCode.ERR_StaticLocalFunctionCannotCaptureVariable, "c").WithArguments("c").WithLocation(7, 7));
+        }
+
+        [Fact]
+        public void StaticLocalFunction_CapturingMethodGroup3()
+        {
+            CreateCompilation("""
+                LocalFunc();
+
+                static void LocalFunc()
+                {
+                    var x = this.MyExtension;
+                    x();
+                    var y = base.MyExtension;
+                    y();
+                }
+
+                internal static class Extensions
+                {
+                    public static string MyExtension(this Program c)
+                        => string.Empty;
+                }
+                """).VerifyDiagnostics(
+                    // (5,13): error CS0026: Keyword 'this' is not valid in a static property, static method, or static field initializer
+                    //     var x = this.MyExtension;
+                    Diagnostic(ErrorCode.ERR_ThisInStaticMeth, "this").WithLocation(5, 13),
+                    // (5,13): error CS8422: A static local function cannot contain a reference to 'this' or 'base'.
+                    //     var x = this.MyExtension;
+                    Diagnostic(ErrorCode.ERR_StaticLocalFunctionCannotCaptureThis, "this").WithLocation(5, 13),
+                    // (7,13): error CS1511: Keyword 'base' is not available in a static method
+                    //     var y = base.MyExtension;
+                    Diagnostic(ErrorCode.ERR_BaseInStaticMeth, "base").WithLocation(7, 13),
+                    // (7,13): error CS8422: A static local function cannot contain a reference to 'this' or 'base'.
+                    //     var y = base.MyExtension;
+                    Diagnostic(ErrorCode.ERR_StaticLocalFunctionCannotCaptureThis, "base").WithLocation(7, 13));
+        }
+
+        [Fact]
+        public void StaticLocalFunction_CapturingMethodGroup4()
+        {
+            CreateCompilation("""
+                static class Extensions
+                {
+                    static void F()
+                    {
+                        LocalFunc();
+                        static void LocalFunc()
+                        {
+                            var x = MyExtension;
+                            x(null);
+                        }
+                    }
+                    static string MyExtension(this object o)
+                        => string.Empty;
+                }
+                """).VerifyDiagnostics();
```

Comment:
VerifyEmitDiagnostics? This is a success scenario, let's verify that to the end.